src/expm_frechet.jl

```diff
@@ -0,0 +1,532 @@
+
+@doc raw"""
+based on
+.. [1] Awad H. Al-Mohy and Nicholas J. Higham (2009)
```

Comment:
Please check how we do references in footnotes.

Reply:
This is fixed - I have generated the doc locally - but couldnt find the doc link generated by CI
Manifolds.expm_frechet
— Function

expm_frechet(A, E)
expm_frechet!(buff, A, E)

Compute Frechet derivative of expm(A) in direction E using algorithm 6.4 of [AlMohyHigham2009]

expm_frechet(A,E)=lim⁡t→01t(exp⁡(A+tE)−exp⁡(A))expm\_frechet(A, E) = \lim_{t\to 0}\frac{1}{t}(\exp(A + tE) - \exp(A))
expm_frechet(A,E)=t→0lim​t1​(exp(A+tE)−exp(A))

For sufficiently small ∣A∣1|A|_1∣A∣1​ norm, we use Padé with an appropriate number of terms ( one of (3, 5, 7, 9, 13)), with 13 terms is the default. Otherwise, we use the formula exp(A) = (exp(A/2^s))^{2^s}, where s is used to scale so 2−s∣A∣12^{-s}|A|_12−s∣A∣1​ is smaller than ell_table_61ell\_table\_61ell_table_61[14] of the lookup table ell_table_61ell\_table\_61ell_table_61 in the code, which comes from table 6.1 in op. cit.

For expm_frechet!expm\_frechet!expm_frechet!, buff is a matrix of size 16*k times k, where kkk is the size of AAA. The returns, (exp(A), dexp(A, E)) are stored in the first two blocks of buff. The remaining blocks are used as temporary storage. This is for algorithms calling expm_frechet repeatedly, where allocation becomes the biggest bottleneck.

Reply:
please use (with `@docs raw"""`to start the string for the string Math formulae as actual math – in markdown that is two backtracks, so something line ``` ``\lVert A \rVert_1`` ``` for the matrix 1-norm in the doc string

Reply:
Thanks - fixed - actually the cut and paste (here and below) somehow corrupted the HTML format - it should looks better on the docs webpage.